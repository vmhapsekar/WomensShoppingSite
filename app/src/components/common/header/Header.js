import React from 'react'
import PropTypes from 'prop-types'
import {map} from 'lodash'
import {Grid, GRow, GCol} from '../../common/grid/Grid'
import {Wrapper, Title, Filter} from './styles'

const Header = ({title, filter, setFilter}) => {
  const selectOptions = map(filter, (option, index) => {
    return <option key={index} value={option.id}>{option.label}</option>
  })
  return (
    <Wrapper>
      <Grid>
        <GRow>
          <GCol xs={12} sm={12} md={4} lg={4}>
            <Title>{title}</Title>
          </GCol>
          <GCol xs={12} sm={12} md={4} lg={4}>
            <Filter id='size-filter' className={'dropdown-style'} onChange={setFilter}>
              { selectOptions }
            </Filter>
          </GCol>
        </GRow>
      </Grid>
    </Wrapper>
  )
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  filter: PropTypes.array.isRequired,
  setFilter: PropTypes.func
}

export default Header
