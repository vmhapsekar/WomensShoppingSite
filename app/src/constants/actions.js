export const PRODUCT_FETCH_ALL_REQUEST = 'PRODUCT/FETCH/REQUEST'
export const PRODUCT_FETCH_ALL_SUCCESS = 'PRODUCT/FETCH/SUCCESS'
export const PRODUCT_FETCH_ALL_FAILURE = 'PRODUCT/FETCH/FAILURE'
