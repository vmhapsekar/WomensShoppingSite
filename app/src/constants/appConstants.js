export const APP_SERVICE_TIMEOUT = 30000
export const SIZE_FILTER = [{id: 0, label: 'Filter by size'},
  {id:1, label:'XS'},
  {id:2, label:'S'},
  {id:3, label:'M'},
  {id:4, label:'L'},
  {id:5, label:'XL'}]