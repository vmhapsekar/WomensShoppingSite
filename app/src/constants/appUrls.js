export const LAUNCH_URL = '/'
export const LANDING_URL = '/womens-top'
export const IMAGES_URL = 'http://localhost:9999/images/'
