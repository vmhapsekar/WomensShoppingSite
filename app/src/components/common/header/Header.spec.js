import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {SIZE_FILTER} from '../../../constants/appConstants'

import Header from './Header.js'

describe('<Header />', () => {
  it('WILL match snapshot: <Header />', () => {
    const wrapper = renderer.create(<Header title="Women's tops" filter={SIZE_FILTER} seetFilter={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
})
