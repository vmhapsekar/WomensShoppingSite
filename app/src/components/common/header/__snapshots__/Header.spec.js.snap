// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> WILL match snapshot: <Header /> 1`] = `
.c2 {
  margin-right: auto;
  margin-left: auto;
  padding-right: 0.9375rem;
  padding-left: 0.9375rem;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.9375rem;
  margin-left: -0.9375rem;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.9375rem;
  padding-left: 0.9375rem;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.9375rem;
  padding-left: 0.9375rem;
}

.c1 {
  max-width: 1164px;
  margin: 0 auto;
}

.c0 {
  padding: 0;
  background-color: #E0FFFF;
}

.c5 {
  color: black;
}

.c7 {
  -webkit-appearance: menulist-button;
  border-color: rgb(153,153,153);
  height: 30px;
  width: 100%;
  margin: 0 auto;
}

@media only screen and (min-width:0em) {
  .c4 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:30em) {
  .c4 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c4 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
    display: block;
  }
}

@media only screen and (min-width:60em) {
  .c4 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
    display: block;
  }
}

@media only screen and (min-width:0em) {
  .c6 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:30em) {
  .c6 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c6 {
    -webkit-flex-basis: 16.666666666666668%;
    -ms-flex-preferred-size: 16.666666666666668%;
    flex-basis: 16.666666666666668%;
    max-width: 16.666666666666668%;
    display: block;
  }
}

@media only screen and (min-width:60em) {
  .c6 {
    -webkit-flex-basis: 16.666666666666668%;
    -ms-flex-preferred-size: 16.666666666666668%;
    flex-basis: 16.666666666666668%;
    max-width: 16.666666666666668%;
    display: block;
  }
}

@media (max-width:992px) {
  .c0 {
    width: 110%;
  }
}

@media (max-width:1170px) {
  .c0 {
    width: 110%;
  }
}

@media (max-width:1600px) {
  .c0 {
    width: 102.5%;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 134%;
  }
}

@media (max-width:480px) {
  .c0 {
    width: 90%;
  }
}

@media (max-width:376px) {
  .c0 {
    width: 90%;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h2
          className="c5"
        >
          Women's tops
        </h2>
      </div>
      <div
        className="c6"
      >
        <h2>
          <select
            className="dropdown-style c7"
            id="size-filter"
            onChange={undefined}
          >
            <option
              value={0}
            >
              Filter by size
            </option>
            <option
              value={1}
            >
              XS
            </option>
            <option
              value={2}
            >
              S
            </option>
            <option
              value={3}
            >
              M
            </option>
            <option
              value={4}
            >
              L
            </option>
            <option
              value={5}
            >
              XL
            </option>
          </select>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
